### PR TITLE
Fix Emu.Stop() with vulkan device enumeration hack

### PR DESCRIPTION
**Pull request by elad335**
Do not attemp to join dev enumeration thread on Emu.Stop().

### CODE REVIEW

**Comment by Margen67** [comment]
*enumeration

**Comment by elad335** [comment]
Used a better solution

**Comment by Nekotekina** [comment]
What's the hurry with this one? Using new/delete is literally never justified in modern C++. I suspect it only needs a trivial detached thread.

File: rpcs3/rpcs3qt/emu_settings.cpp [hunk 1]
```diff
@@ -217,6 +218,8 @@ emu_settings::Render_Creator::Render_Creator()
 		supportsVulkan = device_found;
 		vulkanAdapters = std::move(compatible_gpus);
 	}
+
+	enum_thread.join();
```
Comment (MSuih) [inline rpcs3/rpcs3qt/emu_settings.cpp]: Wouldn't this lock up the emulator if user selected ignore and the thread never finishes?

File: rpcs3/rpcs3qt/emu_settings.cpp [hunk 2]
```diff
@@ -189,7 +189,8 @@ emu_settings::Render_Creator::Render_Creator()
 				}
 			}
 		}
-		std::scoped_lock{mtx}, thread_running = false;
+
+		std::lock_guard{mtx}, thread_running = false;
```
Comment (kd-11) [inline rpcs3/rpcs3qt/emu_settings.cpp]: Uhm, aren't we supposed to move to scoped_lock with newer c++?

Comment (elad335) [reply rpcs3/rpcs3qt/emu_settings.cpp]: dunno, its implementation is more complex, as long as compiled code is the same I can revert it. will test soon.

Comment (elad335) [reply rpcs3/rpcs3qt/emu_settings.cpp]: Nevermind its fine.

**Pull request merged**